third_party/java/arcs/build_defs/arcs_ts_test.bzl

```diff
@@ -0,0 +1,14 @@
+"""Macro for running TypeScript tests with bazel."""
+
+load(":sigh.bzl", "sigh_command")
+
+def arcs_ts_test(name, src, deps, flaky = False):
```

Comment:
Why not add this as another `arcs_tool`? 

Optional: Should we add a `test_only` attribute here, defaulted to `1`?

Reply:
`_test` targets are testonly automatically, I believe. See https://docs.bazel.build/versions/master/be/common-definitions.html

Reply:
The linked document refers to things common to all `rules`. Do they mean all `rules` and `macros`?

Reply:
For example, we don't get `deps` or `data` for free in macros, which is why we've added them explicitly. 

https://docs.bazel.build/versions/master/be/common-definitions.html#common-attributes

Reply:
All BUILD targets ending in `_test` or `-test` are considered `testonly`, regardless of whether they were generated by a rule or a macro.

Here's the error message you get when you try to add a test target created via `arcs_ts_test` as a dep in a non-test target:

```
ERROR: /Users/csilvestrini/dev/arcs/src/tools/tests/BUILD:67:16: in run_in_repo rule //src/tools/tests:variable_proto: non-test target '//src/tools/tests:variable_proto' depends on testonly target '//src/tools/tests:schema-generator-test' and doesn't have testonly attribute set
```